Tests/test_QCPSuperimposer.py

```diff
@@ -19,60 +22,141 @@
-
-x = array([[51.65, -1.90, 50.07],
-         [50.40, -1.23, 50.65],
-         [50.68, -0.04, 51.54],
-         [50.22, -0.02, 52.85]], 'f')
-
-y = array([[51.30, -2.99, 46.54],
-         [51.09, -1.88, 47.58],
-         [52.36, -1.20, 48.03],
-         [52.71, -1.18, 49.38]], 'f')
-
-sup = QCPSuperimposer()
-
-# set the coords
-# y will be rotated and translated on x
-sup.set(x, y)
-
-# do the qcp fit
-sup.run()
-
-# get the rmsd
-rms = sup.get_rms()
-
-# get rotation (right multiplying!) and the translation
-rot, tran = sup.get_rotran()
-
-# rotate y on x manually
-y_on_x1 = dot(y, rot) + tran
-
-# same thing
-y_on_x2 = sup.get_transformed()
-
-
-def simple_matrix_print(matrix):
-    """Simple string to display a floating point matrix
-
-    This should give the same output on multiple systems.  This is
-    needed because a simple "print matrix" uses scientific notation
-    which varies between platforms.
-
-    Only 4 decimal places are used to avoid false test failures due
-    to slight differences in the calculation (e.g. due to different
-    versions of the underlying libraries or the compilation options
-    they used).
-    """
-    return "[%s]" % "\n ".join("[%s]" % " ".join("% 1.4f" % v for v in row)
-                               for row in matrix)
-
-
-# output results
-print(simple_matrix_print(y_on_x1))
-print("")
-print(simple_matrix_print(y_on_x2))
-print("")
-print("%.2f" % rms)
+class QCSuperimposerTest(unittest.TestCase):
+
+	def setUp(self):
+		self.x = array([[51.65, -1.90, 50.07],
+		              [50.40, -1.23, 50.65],
+		              [50.68, -0.04, 51.54],
+		              [50.22, -0.02, 52.85]])
+
+		self.y = array([[51.30, -2.99, 46.54],
+		              [51.09, -1.88, 47.58],
+		              [52.36, -1.20, 48.03],
+		              [52.71, -1.18, 49.38]])
+
+		self.sup = QCPSuperimposer()
+		self.sup.set(self.x,self.y)
+
+	#Public methods
+
+	def testSet(self):
+		"""Test method set(x,y)"""
+		assert(array_equal(self.sup.reference_coords, self.x))
+		assert(array_equal(self.sup.coords, self.y))
+		assert(self.sup.transformed_coords == None)
+		assert(self.sup.rot == None)
+		assert(self.sup.tran == None)
+		assert(self.sup.rms == None)
+		assert(self.sup.init_rms == None)
+
+	def testRun(self):
+		"""Test method run()"""
+		self.sup.run()
+		assert(array_equal(self.sup.reference_coords, self.x))
+		assert(array_equal(self.sup.coords, self.y))
+		assert(self.sup.transformed_coords == None)
+		calc_rot = array([[0.68304939, -0.5227742, -0.51004967],
+ 		                [0.53664482, 0.83293151, -0.13504605],
+ 			        [0.49543503, -0.18147239, 0.84947743]])
+		assert(array_equal(around(self.sup.rot, decimals=3), around(calc_rot, decimals=3)))
+		calc_tran = array([-7.43885125, 36.51522275, 36.81135533])
+		assert(array_equal(around(self.sup.tran, decimals=3), around(calc_tran, decimals=3)))
+		calc_rms = 0.003
+		assert(float(('%.3f' % self.sup.rms)) == calc_rms)
```

Comment:
Ideally this would be ``self.assertEqual(float('%.3f' % self.sup.rms), calc_rms)``

There was also a set of extra brackets for calling ``float``.